src/templates/kit/Duration.js

```diff
@@ -0,0 +1,134 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import PropTypes from 'prop-types'
+import { Field, GU, TextInput, useTheme, useViewport } from '@aragon/ui'
+
+const MINUTE_IN_SECONDS = 60
+const HOUR_IN_SECONDS = MINUTE_IN_SECONDS * 60
+const DAY_IN_SECONDS = HOUR_IN_SECONDS * 24
+
+function Duration({ duration, onUpdate, label }) {
+  const theme = useTheme()
+  const { above } = useViewport()
+
+  // Calculate the units based on the initial duration (in seconds).
+  const [baseDays, baseHours, baseMinutes] = useMemo(() => {
+    let remaining = duration
+
+    const days = Math.floor(remaining / DAY_IN_SECONDS)
+    remaining -= days * DAY_IN_SECONDS
+
+    const hours = Math.floor(remaining / HOUR_IN_SECONDS)
+    remaining -= hours * HOUR_IN_SECONDS
+
+    const minutes = Math.floor(remaining / MINUTE_IN_SECONDS)
+    remaining -= minutes * MINUTE_IN_SECONDS
+
+    return [days, hours, minutes]
+  }, [duration])
```

Comment:
Just wondering, I would have expected a `useEffect()` based on `duration` here, in case it ever gets changed from the outside.

As it is, I don't think the state will change if this happens?

Reply:
I think it does already? That’s how the component receives the user input :-)

If `duration` changes, the `useMemo()` is triggered again, which updates `baseDays`, `baseHours` and `baseMinutes`.

Reply:
But do the `useStates()` get updated? I thought you had to do this separately via a `useEffect()`.

Reply:
No, and yes :thinking: 

Let me check why it works :smile: